connectors-e2e-test/connectors-e2e-test-agentic-ai/src/test/java/io/camunda/connector/e2e/agenticai/a2a/A2aStandaloneTests.java

```diff
@@ -0,0 +1,392 @@
+
+    assertVariablesWithWebhook(zeebeTest);
+  }
+
+  private static void assertVariablesWithWebhook(ZeebeTest zeebeTest) {
+    CamundaAssert.assertThat(zeebeTest.getProcessInstanceEvent())
+        .hasVariableSatisfies(
+            "travelAgentFetchCardResponse",
+            A2aClientResponse.class,
+            A2aStandaloneTests::assertAgentCard)
+        .hasVariableSatisfies(
+            "travelAgentWebhookResponse", A2aTask.class, A2aStandaloneTests::assertTask);
+  }
+
+  private static void assertTask(A2aTask task) {
+    assertThat(task.status().state()).isEqualTo(A2aTaskStatus.TaskState.COMPLETED);
+    assertThat(task.id()).isEqualTo("task-001");
+    assertThat(task.contextId()).isEqualTo("ctx-001");
+    assertThat(task.artifacts()).isNotEmpty();
+    assertThat(task.history()).hasSize(3);
+  }
+
+  private static void assertAgentCard(A2aClientResponse response) {
+    Assertions.assertThat(response.pollingData()).isNotNull();
+    Assertions.assertThat(response.pollingData().id()).isNotBlank();
+
+    Assertions.assertThat(response.result().kind()).isEqualTo("agentCard");
+    Assertions.assertThat(response.result()).isInstanceOf(A2aAgentCard.class);
+    var a2aAgentCard = (A2aAgentCard) response.result();
+    assertThat(a2aAgentCard.skills()).isNotEmpty();
+  }
+
+  private void setUpWireMockStubs() {
+    stubFor(
+        get(urlPathEqualTo("/.well-known/agent-card.json"))
+            .willReturn(
+                aResponse()
+                    .withBody(
+                        testFileContent("travel-agent-card.json")
+                            .get()
+                            .formatted(wireMock.getHttpBaseUrl()))));
+
+    stubFor(
+        post(urlPathEqualTo("/travel-agent"))
+            .withRequestBody(WireMock.containing("message/send"))
+            .willReturn(
+                aResponse()
+                    .withBody(testFileContent("travel-agent-response-submitted.json").get())));
+
+    stubFor(
+        post(urlPathEqualTo("/travel-agent"))
+            .inScenario("polling")
+            .whenScenarioStateIs(STARTED)
+            .withRequestBody(WireMock.containing("tasks/get"))
+            .willReturn(
+                aResponse().withBody(testFileContent("travel-agent-response-working.json").get()))
+            .willSetStateTo("complete-task"));
+
+    stubFor(
+        post(urlPathEqualTo("/travel-agent"))
+            .inScenario("polling")
+            .whenScenarioStateIs("complete-task")
+            .withRequestBody(WireMock.containing("tasks/get"))
+            .willReturn(
+                aResponse()
+                    .withBody(testFileContent("travel-agent-response-completed.json").get())));
+  }
+
+  private void callWebhookEndpointWithDelay(
+      String webhookUrl, String payload, long delayInSeconds) {
+    callWebhookEndpointWithDelay(webhookUrl, payload, Map.of(), delayInSeconds);
+  }
+
+  private void callWebhookEndpointWithDelay(
```

Comment:
🔧 Is this the same logic as in the A2aIntegrationTestSupport, but with headers? Could we maybe consolidate those 2?